mathics/builtin/datentime.py

```diff
@@ -1105,24 +1113,24 @@ class TimeConstrained(Builtin):
         the evaluation continues after the timeout. However, at the end of the evaluation, the function will return '$Aborted' and the results will not affect
         the state of the \Mathics kernel.
 
-        """
 
         # FIXME: these tests sometimes cause SEGVs which probably means
         # that TimeConstraint has bugs.
 
         # Consider testing via unit tests.
-        # >> TimeConstrained[Integrate[Sin[x]^1000000,x],1]
-        # = $Aborted
+        >> TimeConstrained[Pause[5];x,.1]
```

Comment:
The problem with code like this is that now tests are 5 seconds times two longer when most of the time it doesn't take that long. 

And this shows a problem with doctests. If you want to write a good test, then it really doesn't make a good example. 

In Python for example you could write an operation and time it, and _then_ pick a pause number that is _a little bit_ large than the last operation run. 

That is a longer way of saying "Consider testing via unit tests". 

 BTW do we know _why_ there were  SEGV's?

Reply:
Yep: the problem is in sympy. To find it I needed such a onerous test.

Reply:
Ok. Thanks for the investigation. One way to let others know this is to add it in a comment. And if it is onerous, then it is fine to add this in a pytest kind (marked that this is known to SEGV).

Reply:
The choice of 5secs instead of 0.5 allows to me to see if the process was killed or not in real-time. Now, I change it to .5.

Reply:
Or even 0. The other point though while it is great that you are doing all of these experiments as tests, does everyone have to know about the experiments that work and the ones that don't work (which generally happen more often).

I mean there has to be a way to be less noisy but on the other hand let folks have some sense of what's up.

Reply:
Now ``TimeConstrained[Integrate[Sin[x]^1000000,x],1]`` was moved into a pytest